### ISSUE TITLE
[Bug] MRV2 LM-head TP and EPLB collective order hangs idle DP ranks

### ISSUE THREAD

**Issue by freyfwt** [vllm-project/vllm-ascend#17232]
### Description

The QWEN3_235B_PD nightly job hangs on its first AISBench warmup request when Model Runner V2, LM-head tensor parallelism, and EPLB run together across external DP ranks. The decode API returns HTTP 500 after an HCCL timeout, the proxy returns 502, and AISBench reports `warmup failed`.

Failing run: https://github.com/vllm-project/vllm-ascend/actions/runs/35253856165/job/105314153314

### Reproduction conditions

Use vLLM Ascend commit `c7ca0b676b9668535f467b78cff1274a4ddb63b2` with vLLM 0.28.0 and the [QWEN3_235B_PD.yaml configuration at that exact commit](https://github.com/vllm-project/vllm-ascend/blob/c7ca0b676b9668535f467b78cff1274a4ddb63b2/tests/e2e/nightly/multi_node/external_dp/config/QWEN3_235B_PD.yaml). The YAML on current `main` has since changed, so the pinned revision matters.

- Hardware/topology: three 16-NPU Ascend A3 nodes; one prefill node (TP=16, DP=1) and two decode nodes (TP=1, external DP=32, 16 local DP ranks on each node, EP enabled).
- Model: `vllm-ascend/Qwen3-235B-A22B-W8A8` with `--quantization ascend`.
- Decode environment: `VLLM_USE_V1=1`, `VLLM_USE_V2_MODEL_RUNNER=1`, `HCCL_EXEC_TIMEOUT=204`, `HCCL_BUFFSIZE=1024`.
- Decode flags: `--async-scheduling --enable-expert-parallel --enable-eplb --max-model-len 36864 --max-num-batched-tokens 256 --max-num-seqs 28`.
- Decode EPLB: `--eplb-config '{"window_size":4000,"step_interval":200,"num_redundant_experts":0,"use_async":true,"log_balancedness":true,"log_balancedness_interval":1}'`.
- Ascend options: `--additional-config '{"finegrained_tp_config":{"lmhead_tensor_parallel_size":8},"eplb_config":{"load_collection_phase":"all"},"enable_mlapo":true}'`.
- Decode graph: `--compilation-config '{"cudagraph_mode":"FULL_DECODE_ONLY","cudagraph_capture_sizes":[6,12,18]}'`.
- PD transport: `MooncakeConnectorV1`, prefill `kv_producer` (DP=1, TP=16), decode `kv_consumer` (DP=32, TP=1), `use_ascend_direct=true`; the pinned YAML contains the complete connector settings.
- Trigger: run the YAML's AISBench GSM8K chat accuracy case (`batch_size=64`, `temperature=0.6`, `top_p=0.95`, `max_out_len=32768`). It fails during the first warmup request, while one decode DP rank handles work and other ranks run dummy steps. This is the observed reproducer; the individual flags have not been minimized experimentally.

### Observed behavior and diagnosis

The decode logs show DP0 entering an eight-rank LM-head `AllGather` at 17:55:37, while idle DP ranks entered a 32-rank EPLB `AllReduce` starting at 17:55:31. HCCL reports a 204-second collective timeout at 17:59:03. The first warmup request then fails with proxy `Bad Gateway`.

On the active rank, `sample()` calls `compute_logits()` (LM-head collective) before its EPLB step. On idle ranks, the V2 `_dummy_run()` override calls the parent first; the parent advances EPLB before the override calls `compute_logits()`. The two rank paths therefore enter their LM-head and EPLB collectives in opposite order. This ordering explains the timeout and is consistent with the per-rank HCCL traces.

### Expected behavior

Idle and active decode ranks should join LM-head TP before advancing the EPLB step, so the warmup request completes without an HCCL timeout.
